File: api/src/main/java/org/roylance/yaas/services/server/IServerTokenService.java
package org.roylance.yaas.services.server;

import org.roylance.yaas.models.YaasModels;

public interface IServerTokenService {
    YaasModels.UIAuthentication generateToken(final YaasModels.User user);
    YaasModels.UIAuthentication validateUser(final String token);
}

File: redis/src/main/java/org/roylance/yaas/redis/services/IServerTokenService.kt
package org.roylance.yaas.redis.services

import org.roylance.yaas.YaasModels

interface IServerTokenService {
    fun generateToken(user: YaasModels.User): YaasModels.UIAuthentication
    fun validateUser(token: String): YaasModels.UIAuthentication
}